k8s-model-server/README.md

```diff
@@ -0,0 +1,170 @@
+the bucket name must be globally unique.
+```commandline
+gsutil mb gs://<ucket-name>
+``` 
+
+
+## Upload a Model
+You can train an inception model using the instructions from the 
+[TensorFlow documentation](https://www.tensorflow.org/tutorials/image_retraining), or you can download the pre-trained
+[example](inception.tar.gz).
+
+If you downloaded the example decompress it.
+```commandline
+tar -zxpf inception.tar.gz
+```
+
+Use [gsutil cp](https://cloud.google.com/storage/docs/gsutil/commands/cp) to upload the inception model to the cloud 
+storage bucket you created above.
+```commandline
+gsutil cp -r inception gs://<bucket-name>
+```
+
+Use [gsutil_ls](https://cloud.google.com/storage/docs/gsutil/commands/ls) to view the contents of your bucket. You 
+will see that the contents of the model are stored in the `gs://<bucket-name>/inception/1` directory. This is the 
+first version of the model that we will serve.
+
+```commandline
+gsutil ls -r gs://<bucket-name>/ 
+gs://<bucket-name>/inception/:
+gs://<bucket-name>/inception/
+
+gs://<bucket-name>/inception/1/:
+gs://<bucket-name>/inception/1/
+gs://<bucket-name>/inception/1/saved_model.pb
+
+gs://<bucket-name>/inception/1/variables/:
+gs://<bucket-name>/inception/1/variables/variables.data-00000-of-00001
+gs://<bucket-name>/inception/1/variables/variables.index
+```
+
+## Create a Kubernetes Cluster
+
+Use [gcloud container clusters create](https://cloud.google.com/sdk/gcloud/reference/container/clusters/create) to turn 
+up a new Kubernetes cluster. 
+```commandline
+gcloud container clusters create model-serving --machine-type=n1-standard-4
+```
+This will create a 3 Node cluster using the n1-standard-4 machine shape for each Node.
+
+After creation is complete, you can use 
+[gcloud container clusters describe](https://cloud.google.com/sdk/gcloud/reference/container/clusters/describe) to view 
+your cluster.
+
+## Configure kubectl 
+In order to create Kubernetes workloads, you need to configure [kubectl](https://kubernetes.io/docs/user-guide/kubectl/) 
+to communicate with your cluster. Use 
+[gcloud container clusters get-credentials](https://cloud.google.com/sdk/gcloud/reference/container/clusters/describe) 
+to configure kubectl.
+
+```commandline
+gcloud container clusters get-credentials model-serving --zone <your-zone> --project <your-project>
+```
+
+## Modify the Deployment
+The Kubernetes Deployment in this [manifest](manifests/model-server.yaml).
+
+You need to update the Deployment's `spec.template.spec.containers[0].image` to point the image you pushed to GCR above.
+
+```yaml
+containers:
+      - name: model-server
+        image: gcr.io/<your-project>/model-server:1.0
+```
+
+You need to update the Deployment's `spec.template.spec.containers[0].command.args` to point the GCS bucket you created  
+above.
+
+```yaml
+ args:
+        - /usr/bin/tensorflow_model_server
+          --port=9000 --model_name=inception --model_base_path=gs://<your-bucket>/inception/
```

Comment:
What happens if no models exist? Would the server crashloop or watch for updates?

Reply:
Slowly crash loops.

Reply:
So by default, if we install TF Serving, then it will keep crashlooping until a model is made available?